packages/extras/src/lib/components/AnimatedSpriteMaterial/AnimatedSpriteMaterial.svelte

```diff
@@ -0,0 +1,269 @@
+<script lang='ts'>
+  import * as THREE from 'three'
+  import {
+    T,
+    asyncWritable,
+    type AsyncWritable,
+    createRawEventDispatcher,
+    useFrame,
+    watch,
+    useLoader,
+    useParent
+  } from '@threlte/core'
+  import type {
+    AnimatedSpriteProps,
+    AnimatedSpriteEvents,
+    AnimatedSpriteSlots,
+    SpriteJsonHashData,
+    FrameTag,
+    Frame
+  } from './AnimatedSpriteMaterial.svelte'
+  import { useTexture } from '../../hooks/useTexture'
+
+  type $$Props = Required<AnimatedSpriteProps>
+  type $$Events = AnimatedSpriteEvents
+  type $$Slots = AnimatedSpriteSlots
+
+  export let textureUrl: $$Props['textureUrl']
+  export let dataUrl: $$Props['dataUrl'] = ''
+  export let animation: $$Props['animation'] = ''
+  export let loop: $$Props['loop'] = true
+  export let autoplay: $$Props['autoplay'] = true
+  export let transitionOnLoopEnd: $$Props['transitionOnLoopEnd'] = true
+  export let fps: $$Props['fps'] = 10
+  export let filter: $$Props['filter'] = 'nearest'
+  export let alphaTest: $$Props['alphaTest'] = 0.1
+  export let delay: $$Props['delay'] = 0
+  export let transparent: $$Props['transparent'] = true
+  export let flipX: $$Props['flipX'] = false
+  export let startFrame: $$Props['startFrame'] = 0
+  export let endFrame: $$Props['endFrame'] | undefined = undefined
+  export let rows: $$Props['rows'] = 1
+  export let columns: $$Props['columns'] | undefined = undefined
+  export let totalFrames: $$Props['totalFrames'] = 0
+
+  const parent = useParent()
+  const dispatch = createRawEventDispatcher<$$Events>()
+
+  let timerOffset = 0
+  let currentFrame = startFrame
+  let numFrames = 0
+  let flipOffset = flipX ? -1 : 1
+  let frameWidth = 0
+  let frameHeight = 0
+  let texture: THREE.Texture | undefined
+  let json: SpriteJsonHashData | undefined
+  let frameNames: string[] = []
+  let frameTag: FrameTag | undefined
+  let spritesheetSize = { w: 0, h: 0 }
+  let playing = true
+
+
+  let isMesh = 'isMesh' in $parent!
+  $: isMesh = 'isMesh' in $parent!
+  $: fpsInterval = 1000 / fps
+
+  export let is: THREE.Material = isMesh ? new THREE.MeshBasicMaterial() : new THREE.SpriteMaterial()
+  export let ref: THREE.Material
+
+  const textureStore = useTexture(textureUrl, {
+    transform: (value: THREE.Texture) => {
+      value.matrixAutoUpdate = false
+      value.generateMipmaps = false
+      value.premultiplyAlpha = false
+      value.wrapS = value.wrapT = THREE.RepeatWrapping
+      value.magFilter = value.minFilter = filter === 'nearest'
+        ? THREE.NearestFilter
+        : THREE.LinearFilter
+      return value
+    }
+  })
```

Comment:
Add suspense support

```suggestion
  const suspend = useSuspense()

  const textureStore = suspend(useTexture(textureUrl, {
    transform: (value: THREE.Texture) => {
      value.matrixAutoUpdate = false
      value.generateMipmaps = false
      value.premultiplyAlpha = false
      value.wrapS = value.wrapT = THREE.RepeatWrapping
      value.magFilter = value.minFilter = filter === 'nearest'
        ? THREE.NearestFilter
        : THREE.LinearFilter
      return value
    }
  }))
```